extra_data/sourcedata.py

```diff
@@ -70,6 +72,31 @@ def __getitem__(self, key):
             inc_suspect_trains=self.inc_suspect_trains,
         )
 
+    def _get_first_source_file(self):
+        sample_path = self[next(iter(self.keys()))].source_file_paths[0]
+        return FileAccess(sample_path)
+
+    @property
+    def storage_class(self):
+        if self._first_source_file is ...:
+            self._first_source_file = self._get_first_source_file()
+
+        return self._first_source_file.storage_class
+
+    @property
+    def file_prefix(self):
+        if self._first_source_file is ...:
+            self._first_source_file = self._get_first_source_file()
+
+        return self._first_source_file.file_prefix
```

Comment:
Do we have a use case for this one? 'File prefix' seems vague enough that I'm inclined to leave it out.

Reply:
Inspired by the introduction of a data category for files in the SDP draft which can move through storage classes, I would like to use this field for this purpose.

It addresses the problem that using it for the actual storage class would mean renaming the files as they are copied (say from `raw` to `red`) as well as lose a distinction about their origin. At the same time, it can immediately be used to indicate the result of a reduction, say by prefixing the output of a `RAW-` file with `RRAW-`.

We can drop it if you consider it too vague for now, but I have need of it in the data reduction toolchain.